src/main/java/com/qaprosoft/carina/demo/mobile/gui/mfp/pages/android/CustomDashboardPage.java

```diff
@@ -0,0 +1,30 @@
+package com.qaprosoft.carina.demo.mobile.gui.mfp.pages.android;
+
+import com.qaprosoft.carina.core.foundation.utils.factory.DeviceType;
+import com.qaprosoft.carina.core.foundation.webdriver.decorator.ExtendedWebElement;
+import com.qaprosoft.carina.core.foundation.webdriver.locator.ExtendedFindBy;
+import com.qaprosoft.carina.demo.mobile.gui.mfp.pages.common.CustomDashboardPageBase;
+import com.qaprosoft.carina.demo.mobile.gui.mfp.pages.common.DiaryPageBase;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.support.FindBy;
+
+@DeviceType(pageType = DeviceType.Type.ANDROID_PHONE, parentClass = CustomDashboardPageBase.class)
+public class CustomDashboardPage extends CustomDashboardPageBase {
+
+    @FindBy(xpath = "//*[@resource-id='com.myfitnesspal.android:id/%s']")
+    private ExtendedWebElement optionRadioButton;
+
+    @ExtendedFindBy(accessibilityId = "Done")
+    private ExtendedWebElement customSummaryDoneButton;
+
+    public CustomDashboardPage(WebDriver driver) {
+        super(driver);
+    }
+
+    @Override
+    public DiaryPageBase selectCustomDashboard(Options option) {
+        optionRadioButton.format(option.getOptionId()).click(3);
+        if(option.equals(Options.CUSTOM)) customSummaryDoneButton.click(3);
+        return initPage(getDriver(), DiaryPageBase.class);
```

Comment:
you'll have to change the logic of this method - you'll see why in testcase 5. before if - good decision with id in enum